### PR TITLE
Add option to use a remote debugger proxy

### PR DESCRIPTION
**Pull request by ThomasFettSauce**
## Proposed changes

The changes make it possible to let Appium's remote debugger connect to a proxy instead of the iOS simulator directly when running tests on iOS simulators.
This enables sending additional Web Inspector protocol messages to the simulator and also logging all Web Inspector protocol messages that are sent from Appium to the simulator and vice versa.

The actual implementation and message handling of the proxy is completely up to the developers who want to use this feature.

The pull requests for the other affected Appium packages of this feature are:
* https://github.com/appium/appium-xcuitest-driver/pull/710
* https://github.com/appium/appium-remote-debugger/pull/81

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/parser.js [hunk 1]
```diff
@@ -350,6 +350,27 @@ const args = [
     help: '(IOS-only) Local port used for communication with ios-webkit-debug-proxy'
   }],
 
+  [['--remote-debug-proxy-port'], {
+    defaultValue: false,
+    dest: 'remoteDebugProxyPort',
+    required: false,
+    type: 'int',
+    example: "12000",
+    help: '(IOS-only, <= 11.2) If set, Appium sends and receives remote debugging messages' +
```
Comment (imurchie) [inline lib/parser.js]: As implemented in the `appium-remote-debugger`, this will only work for simulators. That should be mentioned here and below.

**Review by imurchie — COMMENTED** [review]
Why server flags rather than desired capabilities?

File: lib/parser.js (hunk 1 above)
Comment (ThomasFettSauce) [reply lib/parser.js]: Thanks for the feedback! I wasn't aware that desired capabilities just work out of the box. I removed the server flags and added them as desired capabilities.

**Comment by KazuCocoa** [comment]
Is it good to add them to https://github.com/appium/appium-ios-driver/blob/f6e335adf3d1d0c450df378aee9e1b85c90ee1e8/lib/desired-caps.js#L99 to validate them?

**Comment by imurchie** [comment]
Yes, the final step for the desired capability is to add them as @KazuCocoa noted above. Without that, there will be a line in the logs saying that the cap is not recognized, and also the type of value would be unknown.

File: docs/en/writing-running-appium/caps.md [hunk 2]
```diff
@@ -144,6 +144,8 @@ Driver](/docs/en/drivers/ios-uiautomation.md).
 |`appName`| The display name of the application under test. Used to automate backgrounding the app in iOS 9+.|e.g., `UICatalog`|
 |`customSSLCert`|(Sim only) Add an SSL certificate to IOS Simulator. | e.g. <br/>`-----BEGIN CERTIFICATE-----MIIFWjCCBEKg...`<br/>`-----END CERTIFICATE-----`|
 |`webkitResponseTimeout`|(Real device only) Set the time, in ms, to wait for a response from WebKit in a Safari session. Defaults to `5000`|e.g., `10000`|
+|`remoteDebugProxyPort`| (Sim only, <= 11.2) If set, Appium sends and receives remote debugging messages through a proxy on this local port instead of communicating with the iOS remote debugger directly. |e.g. `12000`|
```
Comment (jlipps) [inline docs/en/writing-running-appium/caps.md]: what about having just one capability `remoteDebugProxy` which could take either a port or a socket?

**Comment by imurchie** [comment]
Everything is available in the drivers now, so merging. Thanks!

**Pull request merged**